File: pages/api-drivers/drivers-connectors/code-create-connection.py
# Two ways to connect using the Python driver:

# 1. Automatic connection

from trueno_python_driver import Trueno

trueno = Trueno(host='http://localhost', port=8000)

#May use 'with'
with Trueno(host='http://localhost', port=8000) as trueno:
    if trueno.connected():
       #Do work

#######

# 2. Manual connection (can define callbacks)

from trueno_python_driver import Trueno

def conn_callback(*args):
    print('Connected')

def disc_callback(*args):
    print('Disconnected')


trueno = Trueno(host='http://localhost', port=8000, auto_connect=False)

trueno.connect(conn_callback, disc_callback)

File: docs/latest/docs/latest/docs/latest/pages/api-drivers/drivers-connectors/code-create-connection.js
/* require if in node */
const Trueno = require('trueno-javascript-driver');

/* create driver instance */
let trueno = new Trueno({host: 'http://localhost', port: 8000});

/* connecting to trueno backend */
trueno.connect((s)=> {
  /* connected, s is a socket */
  console.log('connected', s.id);
}, ()=> {
  /* disconnected from trueno backend */
  console.log('disconnected');
});